hw/ip/otp_ctrl/util/crypto_present.py

```diff
@@ -0,0 +1,65 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# default key size 128 bits, input data 64 bits
+from ctypes import c_uint64, c_uint8, CDLL
+
+lib_path = "../../prim/dv/prim_present/crypto_dpi_present/crypto_present.so"
+num_round = 31
+
+
+def load_present_c_lib():
+    '''Load crypto_present.so library and cast input and output with ctypes'''
+    try:
+        present_lib = CDLL(lib_path)
+    except Exception as e:
+        print(e, ':could not recogized lib path: ', lib_path)
+
+    present_lib.c_encrypt.restype = c_uint64
+    present_lib.c_encrypt.argtypes = (c_uint64, c_uint64, c_uint64, c_uint8, c_uint8)
+    present_lib.c_decrypt.restype = c_uint64
+    present_lib.c_decrypt.argtypes = (c_uint64, c_uint64, c_uint64, c_uint8, c_uint8)
+
+    return present_lib
+
+
+def get_keys(key):
+    '''Separate keys to upper 64 bits and lower 64 bits'''
+    return key >> 64, key % (1 << 64)
+
+
+def present_encrypt(plaintext, key):
```

Comment:
to make the interface consistent, should we maybe expect these to be hex-strings as well? in that case, I believe these values need to be converted to integers first via `int(x, 16)` casts.

Reply:
I tried hex or int as input, i think they both works, because the input is actually going to cast to c_uint64 so i think either input is fine.
For hex input, I used the format like `0x0ffdd1122`, is that what you are looking for?

Reply:
ah I see. I thought the `hex()` cast does not prefix the string with `0x`, but it looks like it does. so then this fine since both the inputs and output can be in the same format.